octopoes/tools/xtdb_client.py

```diff
@@ -119,7 +119,9 @@ def tx_log(
         return res.json()
 
     def submit_tx(self, transactions: list[str]) -> JsonValue:
-        res = self._client.post("/submit-tx", json={"tx-ops": transactions})
+        joined_transactions = ", ".join(transactions)
+        data = f'{{"tx-ops": [{joined_transactions}]}}'
+        res = self._client.post("/submit-tx", content=data, headers={"Content-Type": "application/json"})
```

Comment:
It would be preferable to use the `json` parameter of the `post` method, as it automatically serializes the request body and sets the appropriate `Content-Type` header. This improves readability and maintainability, as working with (in this case) dictionaries is generally more straightforward than working with strings to generate JSON.

```suggestion
        data = {"tx-ops": transactions}
        res = self._client.post("/submit-tx", json=data)
```

It also doesn't necessarily have to perform much worse than the current implementation, since *serialization* is usually quite fast in this context (the other way, *deserializing*, would be a different story). I think this compensates for readability and maintainability, since it's easier to work with these data types than strings.